src/main/java/org/terasology/dialogs/action/NewDialogActionTypeHandler.java

```diff
@@ -17,33 +17,24 @@
 package org.terasology.dialogs.action;
 
 import java.util.Map;
+import java.util.Optional;
 
-import org.terasology.persistence.typeHandling.DeserializationContext;
-import org.terasology.persistence.typeHandling.PersistedData;
-import org.terasology.persistence.typeHandling.PersistedDataMap;
-import org.terasology.persistence.typeHandling.RegisterTypeHandler;
-import org.terasology.persistence.typeHandling.SerializationContext;
-import org.terasology.persistence.typeHandling.SimpleTypeHandler;
+import org.terasology.persistence.typeHandling.*;
 
 import com.google.common.collect.ImmutableMap;
 
 @RegisterTypeHandler
-public class NewDialogActionTypeHandler extends SimpleTypeHandler<NewDialogAction> {
+public class NewDialogActionTypeHandler extends TypeHandler<NewDialogAction> {
 
     @Override
-    public PersistedData serialize(NewDialogAction action, SerializationContext context) {
-        Map<String, PersistedData> data = ImmutableMap.of(
-                "type", context.create(action.getClass().getSimpleName()),
-                "target", context.create(action.getTarget()));
-
-        return context.create(data);
+    public PersistedData serializeNonNull(NewDialogAction action, PersistedDataSerializer context) {
+        return context.serialize(action.getTarget());
     }
 
     @Override
-    public NewDialogAction deserialize(PersistedData data, DeserializationContext context) {
+    public Optional<NewDialogAction> deserialize(PersistedData data) {
         PersistedDataMap root = data.getAsValueMap();
         String target = root.get("target").getAsString();
```

Comment:
Same as above -- change deserialization code to use the format in which this type of value is serialized, and use failing conditions to prevent exceptions.